app/helpers/products_helper.rb

```diff
@@ -0,0 +1,20 @@
+module ProductsHelper
+  def recently_ids
+    cookies[:recently_ids] ||= Array.new
+  end
+
+  def add_recently_id
+    list_id = if recently_ids
+                recently_ids.split("&")
+              else
+                recently_ids
+              end
+    list_id.unshift(@product.id) unless list_id.include?(@product.id.to_s)
+    list_id.pop if list_id.length > Settings.product.recently_size
+    cookies.permanent[:recently_ids] = list_id
+  end
+
+  def find_recently_products
+    Product.list_by_ids(recently_ids.split("&"))
```

Comment:
`recently_ids` -> nil thì sao

nên dùng toán tử an toàn nhỉ

Reply:
Em nghĩ chỗ này không cần chị à. recently_ids = cookies[:recently_ids] ||= Array.new . nên nó không nil được chị . nếu cookies[:recently_ids] = nil thì recently_ids sẽ là 1 mảng rỗng . vậy gọi method  find_recently_products bị rỗng thì bên view hiển thị không có sản phẩm là được chị